skill_tagging/skill_tagging_mixin.py

```diff
@@ -0,0 +1,82 @@
+"""
+A mixin that fetches and verifies skills related to an Xblock, that can be added for all XBlocks.
+"""
+
+import logging
+from django.conf import settings
+from django.utils.translation import gettext as _
+from openedx_events.learning.data import XBlockSkillVerificationData
+from openedx_events.learning.signals import XBLOCK_SKILL_VERIFIED
+from urllib.parse import quote, urljoin
+from xblock.core import XBlock
+from xblock.fields import Boolean, Scope
+
+from .utils import get_api_client
+
+LOGGER = logging.getLogger(__name__)
+
+# Make '_' a no-op so we can scrape strings
+_ = lambda text: text
+
+
+@XBlock.wants('user')
+class SkillTaggingMixin:
+    has_verified_tags = Boolean(
+        display_name=_("Has verified tags"),
+        default=False,
+        help=_("Xblock has verified tags?"),
+        scope=Scope.user_state
+    )
+    
+    def _fetch_skill_tags(self):
+        if hasattr(settings, 'TAXONOMY_API_BASE_URL'):
+            API_BASE_URL = settings.TAXONOMY_API_BASE_URL
+        else:
+            API_BASE_URL = None
+
+        if not API_BASE_URL:
+            LOGGER.warning("Require TAXONOMY_API_BASE_URL to be present in the settings.")
+            return
```

Comment:
NIT: we can skip creating `API_BASE_URL` and directly use `settings.TAXONOMY_API_BASE_URL` after this check.
```suggestion
        if not hasattr(settings, 'TAXONOMY_API_BASE_URL'):
            LOGGER.warning("Require TAXONOMY_API_BASE_URL to be present in the settings.")
            return
```

Reply:
Updated.